library/std/src/sys/pal/unix/thread.rs

```diff
@@ -803,12 +803,30 @@ pub mod guard {
         Some(stack_ptr.with_addr(stackaddr))
     }
 
+    #[cfg(target_os = "netbsd")]
+    unsafe fn get_stack_start() -> Option<*mut libc::c_void> {
+        let mut ret = None;
+        let mut attr: libc::pthread_attr_t = crate::mem::zeroed();
+        let e = libc::pthread_getattr_np(libc::pthread_self(), &mut attr);
+        if e == 0 {
+            let mut stackaddr = crate::ptr::null_mut();
+            let mut stacksize = 0;
+            let mut guardsize = 0;
+            assert_eq!(libc::pthread_attr_getstack(&attr, &mut stackaddr, &mut stacksize), 0);
+            // on netbsd, we need to take in account the guard size to push up
+            // the stack's address from the bottom.
+            assert_eq!(libc::pthread_attr_getguardsize(&attr, &mut guardsize), 0);
+            stackaddr = stackaddr.add(guardsize);
```

Comment:
Hmm. Is this seriously how NetBSD handles it? It's nonintuitive to me that this is correct in the presence of `setstack` and `setguardsize`. e.g. What if someone calls `pthread_attr_setstack` earlier?

Mmm, I love that BUGS section.

https://man.netbsd.org/pthread_attr_getstack.3#BUGS

But seriously, why doesn't NetBSD return the stack address _per se_? This doesn't quite make sense. Stacks "grow down", so if the guard size is N, this informs the _allocated size_, doesn't it, not the _address_?

Reply:
In other words, a stackaddr P would have its final terminus be `((P - stacksize) + guardsize)`, no? But... those bugs...

Reply:
thanks for the feedback. Good points, I ll explore your suggestions later on. Cheers.

Reply:
> FYI, the pthread stackaddr, in the sense of pthread_attr_getstack/setstack, is always the numerically least address that the pthread is guaranteed access to.

Ah, I seem to have gotten things backward, then. I may have misinterpreted some of the OS-specific documentation of this function, and didn't think to check the POSIX docs... it is common for the POSIX docs to be unhelpfully vague and the OS-specific documentation to override it in five different ways, so it's rather charmingly different this time!

Reply:
> On a machine where stacks grow down (which is not all of them), 

I am aware of this being hypothetically permitted, but in practice, I am not aware of Rust supporting the standard library for any targets where a stack-grows-up ABI is permitted or meaningful. Even when the CPU permits, the OS bans.